proxy/src/main/java/com/velocitypowered/proxy/connection/registry/DimensionRegistry.java

```diff
@@ -0,0 +1,128 @@
+package com.velocitypowered.proxy.connection.registry;
+
+import java.util.HashSet;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import javax.annotation.Nonnull;
+
+import net.kyori.nbt.CompoundTag;
+import net.kyori.nbt.ListTag;
+import net.kyori.nbt.Tag;
+import net.kyori.nbt.TagType;
+
+public class DimensionRegistry {
```

Comment:
Can we make this class immutable if possible?

Reply:
Fair, all of the three classes should be. Will do.